Source/Transform/KTFractionalFFT.cc

```diff
@@ -0,0 +1,278 @@
+
+        return true;
+    }
+
+    bool KTFractionalFFT::Initialize(unsigned s)
+    {
+        if( fInitialized )
+        {
+            return true;
+        }
+
+        fForwardFFT.SetComplexAsIQ( true );
+        fForwardFFT.SetTimeSize( s );
+        fForwardFFT.SetTransformFlag( fTransformFlag );
+        fReverseFFT.SetTimeSize( s );
+        fReverseFFT.SetTransformFlag( fTransformFlag );
+
+        if (! fForwardFFT.InitializeForComplexTDD())
+        {
+            KTERROR(evlog, "Error while initializing the forward FFT");
+            return false;
+        }
+
+        if (! fReverseFFT.InitializeForComplexTDD())
+        {
+            KTERROR(evlog, "Error while initializing the reverse FFT");
+            return false;
+        }
+
+        fInitialized = true;
+        return true;
+    }
+
+    bool KTFractionalFFT::ProcessTimeSeriesChirpOnly( KTTimeSeriesData& tsData, KTSliceHeader& slice )
+    {
+        KTDEBUG(evlog, "Receiving time series for fractional FFT");
+
+        KTTimeSeriesFFTW* ts = nullptr;         // time series from data
+        
+        double norm = 0.;                       // norm of the current TS value
+        double phase = 0.;                      // argument of current TS value
+        double binOffset = slice.GetTimeInAcq() / (double)slice.GetBinWidth();
+        double q = fSlope * KTMath::Pi() * slice.GetBinWidth() * slice.GetBinWidth();
+
+        for( unsigned iComponent = 0; iComponent < tsData.GetNComponents(); ++iComponent )
+        {
+            KTDEBUG(evlog, "Processing component: " << iComponent);
+
+            // get TS from data object and make the new TS
+            ts = dynamic_cast< KTTimeSeriesFFTW* >(tsData.GetTimeSeries( iComponent ));
+            if( ts == nullptr )
+            {
+                KTWARN(evlog, "Couldn't find time series object. Continuing to next component");
+                continue;
+            }
+            
+            // Loop through all bins
+            for( unsigned iBin = 0; iBin < ts->GetNTimeBins(); ++iBin )
+            {
+                // Obtain norm and phase from components
+                norm = std::hypot( (*ts)(iBin)[0], (*ts)(iBin)[1] );
+                phase = atan2( (*ts)(iBin)[1], (*ts)(iBin)[0] );
+
+                // Shift phase
+                phase -= q * ((double)iBin + binOffset) * ((double)iBin + binOffset);
+
+                // Assign components from norm and new phase
+                (*ts)(iBin)[0] = norm * cos( phase );
+                (*ts)(iBin)[1] = norm * sin( phase );
+            }
+        }
+
+        return true;
+    }
+
+    bool KTFractionalFFT::ProcessTimeSeries( KTTimeSeriesData& tsData, KTTimeSeriesData& newTSData, KTFrequencySpectrumDataFFTW& newFSData, KTSliceHeader& slice )
+    {
+        KTDEBUG(evlog, "Receiving time series for fractional FFT");
+
+        KTTimeSeriesFFTW* ts = nullptr;   // time series from data
```

Comment:
Done. q1 and q2 I left named that way because they have a different role compared to only q in the previous method, although the operation is the same